File: Medium/0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string rle(string s){
        char curr = s[0];
        int cnt = 0;
        string ans = "";
        for(auto c: s){
            if(c == curr)
                cnt++;
            else{
                ans += to_string(cnt) + curr;
                curr = c;
                cnt = 1;
            }
        }
        ans += to_string(cnt) + curr;
        return ans;
    }
    string countAndSay(int n) {
        string curr = "1";
        for(int i=1; i<n; i++)
            curr = rle(curr);
        return curr;
    }
};

File: Medium/0038-count-and-say/0038-count-and-say.py
class Solution:
    def rle(self, s):
        cnt = 0
        curr = s[0]
        ans = ""
        for c in s:
            if(c == curr):
                cnt += 1
            else:
                ans += str(cnt) + curr
                curr = c
                cnt = 1
        ans += str(cnt) + curr
        return ans

    def countAndSay(self, n: int) -> str:
        curr = "1"
        for i in range(1, n):
            curr = self.rle(curr)
        return curr